File: app/src/main/java/com/polok/eubmanagement/presentation/schedule/scheduleadd/ScheduleAddFragment.java
package com.polok.eubmanagement.presentation.schedule.scheduleadd;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import androidx.lifecycle.ViewModelProvider;
import com.polok.eubmanagement.R;
import com.polok.eubmanagement.base.BaseFragment;
import com.polok.eubmanagement.base.BaseViewModel;
import com.polok.eubmanagement.databinding.FragmentScheduleAddBinding;
import com.polok.eubmanagement.util.SharedPref;
import com.polok.eubmanagement.widget.PrimaryLoader;

public class ScheduleAddFragment extends BaseFragment<FragmentScheduleAddBinding> {
    @Override
    protected FragmentScheduleAddBinding initViewBinding() {
        return FragmentScheduleAddBinding.inflate(getLayoutInflater());
    }
    ScheduleAddViewModel viewModel;
    @Override
    protected BaseViewModel initViewModel() {
        return viewModel = new ViewModelProvider(this).get(ScheduleAddViewModel.class);
    }
    @Override
    protected void initOnCreateView(Bundle savedInstanceState) {
        viewModel.dayZero = getResources().getStringArray(R.array.day_spinner)[0];
        binding.daySpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> adapterView, View view, int pos, long l) {
                viewModel.day = getResources().getStringArray(R.array.day_spinner)[pos];
            }
            @Override
            public void onNothingSelected(AdapterView<?> adapterView) {}
        });

        binding.addScheduleButton.setOnClickListener(view -> {
            SharedPref.init(getContext());
            viewModel.validateScheduleInputAndUploadToFirebase(
                    binding.courseTitle, binding.courseCode, binding.lecturerName, binding.startEndTime, binding.roomNo
            );
        });
    }
    @Override
    protected void onNavigateEvent(int id, Bundle bundle) {
        super.onNavigateEvent(id, bundle);
        if (id == 1) getActivity().onBackPressed();
    }
    @Override
    protected PrimaryLoader initPrimaryLoader() {return binding.primaryLoader;}
}

File: app/src/main/java/com/polok/eubmanagement/presentation/schedule/scheduleadd/ScheduleAddFragment.kt
package com.polok.eubmanagement.presentation.schedule.scheduleadd

import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import androidx.fragment.app.viewModels
import com.polok.eubmanagement.R
import com.polok.eubmanagement.base.BaseFragment
import com.polok.eubmanagement.base.BaseViewModel
import com.polok.eubmanagement.databinding.FragmentScheduleAddBinding
import com.polok.eubmanagement.util.SharedPref
import com.polok.eubmanagement.widget.PrimaryLoader

class ScheduleAddFragment : BaseFragment<FragmentScheduleAddBinding>(
    viewBindingFactory = FragmentScheduleAddBinding::inflate
) {
    private val viewModel: ScheduleAddViewModel by viewModels {
        ScheduleAddViewModel.Factory()
    }

    override fun initViewModel(): BaseViewModel = viewModel

    override fun initOnCreateView(savedInstanceState: Bundle?) {

        viewModel.dayZero = resources.getStringArray(R.array.day_spinner)[0]

        binding.daySpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(
                adapterView: AdapterView<*>?, view: View, pos: Int, l: Long
            ) {
                viewModel.day = resources.getStringArray(R.array.day_spinner)[pos]
            }

            override fun onNothingSelected(adapterView: AdapterView<*>?) {}
        }

        binding.submitButton.setOnClickListener {
            SharedPref.init(context)
            viewModel.validateScheduleInputAndUploadToFirebase(
                binding.courseTitle,
                binding.courseCode,
                binding.lecturerName,
                binding.startEndTime,
                binding.roomNo
            )
        }
    }

    override fun initPrimaryLoader(): PrimaryLoader = binding.primaryLoader

    override fun onNavigateEvent(id: Int, bundle: Bundle?) {
        super.onNavigateEvent(id, bundle)
        if (id == 0) activity?.onBackPressedDispatcher?.onBackPressed()
    }
}